[docs][ENG-4933] add information about CocoaPods cache server (#19050)
expo/expo: docs/pages/build-reference/caching.md

## iOS dependencies

> EAS Build uses a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs and makes the service
> more resistant to the CocoaPods CDN outages.

Reviewer: ```suggestion
EAS Build runs a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs. It also makes the service more resilient to CocoaPods CDN outages.
```

> Currently we are caching all the pods available to be fetched from the CocoaPods CDN.

Reviewer: ```suggestion
Currently, EAS Build is configured to cache all the pods fetched from the CocoaPods CDN.
```

- `jcenter` - [https://jcenter.bintray.com/](https://jcenter.bintray.com/)
- `plugins` - [https://plugins.gradle.org/m2/](https://plugins.gradle.org/m2/)

To disable using our Maven cache server for your builds set `EAS_BUILD_DISABLE_MAVEN_CACHE` env variable value to `"1"` in `eas.json`.

```json
{
  "build": {
    "production": {
        "env": {
            "EAS_BUILD_DISABLE_MAVEN_CACHE": "1",
            // ...
        }
        // ...
    }
    // ...
  }
  // ...
}
```

## iOS dependencies

EAS Build runs a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs. It also makes the service more resilient to CocoaPods CDN outages.

Currently, EAS Build is configured to cache almost all the pods served from official CocoaPods CDN, minus few exceptions which can't be handled by our cache server. These exceptions can be found on the [blacklist](https://github.com/expo/eas-build/blob/main/packages/cocoapods-nexus-plugin/lib/cocoapods_plugin.rb#L6) of our custom CocoaPods plugin and are fetched directly from CDN instead of being fetched through our cache server.

> We also cache `Podfile.lock` in order to provide consistent results across managed app builds.

Reviewer 2: ```suggestion
We also cache `Podfile.lock` to provide consistent results across managed app builds.
```
